source/server/session_resource_repository.h

```diff
@@ -133,7 +139,13 @@ int SessionResourceRepository<TResourceHandle>::add_dependent_session(
   // cleanup code will be executed when the initiating session is closed.
   // We're only responsible for cleaning up the map structures.
   auto status = add_session(
-      session_name, init_func, [](uint32_t id) {}, session_id, /* include_in_reverse_lookup = */ false);
+      session_name,
+      init_func,
+      [](uint32_t id) {},
+      session_id,
+      /* initialization_behavior = */ SESSION_INITIALIZATION_BEHAVIOR_UNSPECIFIED,
+      /* initialized_new_session = */ nullptr,
```

Comment:
Much further down there's a local member `added_new_handle_` which seems to be tracking exactly the same thing. Should getting rid of that duplication be a separate PR?

Reply:
Oh, you're right. I forgot we already had that bookkeeping for another reason. I can make a follow up PR to try to just return that instead.

Reply:
Agreed, it would only pollute this PR. That cleanup would be tech debt, and this one is needed ASAP for its functionality.

Reply:
Sounds good. Added this AB#2191089 tech debt to tackle it separately and assigned it to myself. I just put it in the tech debt bucket but can move it elsewhere if needed.